plugins/module_utils/common.py

```diff
@@ -708,6 +708,11 @@ def perform_action(self, resource, definition):
                         if self.warnings:
                             msg += "\n" + "\n    ".join(self.warnings)
                         self.fail_json(msg=msg, error=exc.status, status=exc.status, reason=exc.reason)
+                    except Exception as exc:
+                        msg = "Failed to create object: {0}".format(exc)
```

Comment:
If we only catch the `ValueError`, we can show up a better message to the user. e.g: Missing or invalid parameter.